core/test/mpi/distributed/row_gatherer.cpp

```diff
@@ -76,6 +76,19 @@ TYPED_TEST(RowGatherer, CanDefaultConstruct)
 }
 
 
+TYPED_TEST(RowGatherer, HasDefaultCollectiveCommunicator)
+{
+    using RowGatherer = typename TestFixture::row_gatherer_type;
+
+    auto rg = RowGatherer::create(this->ref, this->comm);
+
+    auto coll_comm = rg->get_collective_communicator();
+    ASSERT_NE(std::dynamic_pointer_cast<
```

Comment:
you could also just use `gko::as` and let it throw in the error case. No strong preference though